### ISSUE TITLE
Input and output file support flow?

### ISSUE THREAD

**Issue by wang19951120** [a-schild/jave2#69]
Such as inputstream outputstream

**Comment by wang19951120** [comment]
Is the video blocked during the conversion process? Can I convert multiple colleagues?

**Comment by wang19951120** [comment]
Thank you for your help，I may be able to provide meager code and sample support for my project during use.

**Comment by a-schild** [comment]
@wang19951120 yes, if you have some (mostly) working example, I might be able to integrate it in the library

**Comment by tengjianlei** [comment]
when I used the command ffmpeg -i https://vedio on linux os,it throws an error named Input/output error.
for example:ffmpeg version 4.1.3-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 6.3.0 (Debian 6.3.0-18+deb9u1) 20170516
  configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc-6 --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzvbi --enable-libzimg
  libavutil      56. 22.100 / 56. 22.100
  libavcodec     58. 35.100 / 58. 35.100
  libavformat    58. 20.100 / 58. 20.100
  libavdevice    58.  5.100 / 58.  5.100
  libavfilter     7. 40.101 /  7. 40.101
  libswscale      5.  3.100 /  5.  3.100
  libswresample   3.  3.100 /  3.  3.100
  libpostproc    55.  3.100 / 55.  3.100
[tcp @ 0x68a37c0] Failed to resolve hostname oss.cn-north-3.inspurcloudoss.com: Name or service not known
http://oss.cn-north-3.inspurcloudoss.com/vod-cn-north-3-4a41da76-2d73-47a3-b676-9c089f190294/1.SampleAudio_0.4mb5.mp3: Input/output error

**Comment by wangwei91** [comment]
Hello, I come across a scenario where the audio file is transferred from page to page, and Java is required to convert it into binary in wav format, without storage. Can I support this? 
Thank you

**Comment by a-schild** [comment]
Not at the moment, and input source URL is accepted, but the result is currently always a file.
Perhaps if you can write/create some wrapper which makes the input source into an url, then this could work

**Comment by a-schild** [comment]
Gathering this together with the other requests for the same thing into #283.

Six issues have asked for streams or byte arrays as input and output since 2018, each with part of the picture. #283 lists what was asked across all of them, describes why it is more involved than an extra overload — non seekable output breaks the mp4 muxer, an unseekable source changes what ffmpeg can probe, and feeding stdin while draining stdout needs a pump thread or it deadlocks — and carries a workaround for anyone who needs one today.

Closing here so the discussion happens in one place. Nothing is dropped, and your request is quoted there.
